src/electronApp/angularApp/components/connect-form/connect-form.component.ts

```diff
@@ -72,9 +104,19 @@ export class ConnectFormComponent implements OnInit {
    * Invoked when the Angular component is initialized.
    */
   ngOnInit(): void {
-    this.route.queryParams.subscribe(params => {
+    this.route.queryParams.subscribe((params) => {
       this.returnUrl = params.returnUrl as string;
     });
+    this.ips = DataSaver.GetSavedIPs();
+    let lastIP = this.ips[0];
+    this.PrinterAddress.setValue(lastIP);
+  }
+
+  /**
+   * Invoked when the Angular component finished rendering.
+   */
+  ngAfterViewInit(): void {
```

Comment:
How come this is needed?

Reply:
This is needed to focus on the input field. If I just use the autofocus tag on the HTML element it only focuses the first time the connect screen is shown (when the application is opened). And if the click is set onInit the UI doesn't update accordingly.
But I'm not really happy with the current implementation as the input looks like this.
![Input field autofocus (looks like it's not active)](https://user-images.githubusercontent.com/40121865/108602145-beeb8600-73a0-11eb-929a-f5c163e41c0f.png)
and not active like when I click with my mouse:
![Clicked input field (looks active)](https://user-images.githubusercontent.com/40121865/108602170-e5112600-73a0-11eb-8531-4c2de21b55f8.png)

I think I should implement this functionality with `autofocus` but I don't know why it only works on the first try. Maybe it's the way how the routing works.

This is how autofocus currently behaves:
![The input field is focused on the first time it's shown, but not after that](https://user-images.githubusercontent.com/40121865/108602543-beec8580-73a2-11eb-9c61-5187a0f4c0d8.gif)

Reply:
Looks like the IP list had something to do with this autofocus problem.

In this gif, I removed the IP list and added the autocomplete functionality according to [this example from Angular Material](https://stackblitz.com/angular/rmplyklknej?file=src%2Fapp%2Fautocomplete-auto-active-first-option-example.ts). The autofocus is done with `ngAfterViewInit()`. The input field is now correctly highlighted, but the autocomplete dropdown doesn't show up until the field is clicked or written to. The `autofocus` tag on the HTML element would show this, but again only the first time.
![The input field is auto-focused and displayed correctly, but the autocomplete panel doesn't show up](https://user-images.githubusercontent.com/40121865/109206856-acaa8700-77a8-11eb-832b-28e087805c82.gif)

And somehow with this change, the IPs aren't saved between startups, when running with `npm run electron` or after that with `electron .` but if I build the app and install it, the IPs are saved.

Reply:
Fixed the autofocus with 776de02.
Based on [this answer](https://stackoverflow.com/questions/41873893/angular2-autofocus-input-element/42745350#42745350) from StackOverflow.